VMBackup/main/parameterparser.py

```diff
@@ -116,12 +117,17 @@ def __init__(self, protected_settings, public_settings, backup_logger):
                 for config in self.dynamicConfigsFromCRP:
                     if CommonVariables.key in config and CommonVariables.value in config:
                         if((config[CommonVariables.key]).lower() in settingKeysMapping):
-                            self.wellKnownSettingFlags[settingKeysMapping[config[CommonVariables.key].lower()]] = config[CommonVariables.value]
+                            wellKnownSettingFlags[settingKeysMapping[config[CommonVariables.key].lower()]] = config[CommonVariables.value]
```

Comment:
If CRP does not send this dynamicConfigsFromCRP dictionary, then the settings dictionary will not even be having the default values it will be empty when passed to BHS.

Reply:
You mean to say, The intention is to send all well known flags to BHS. If CRP does not send a flag, we will use default value for it in extension? I think that is fair but the implementation of it can be better.

We need to consider that talking to BHS is not the only functionality for the extension. There might be a flag that is only known to CRP & extension and unknown to BHS. In such cases, sending all flags from CRP into BHS is not the best.

So, lets break down the functionality in to two.
1. parameterparser: takes care of receiving dynamicConfigsFromCRP and initiates wellKnownFlags dictionary with appropriate value
2. hostsnapshotter: receives wellKnownFlags as parameter and takes care of translating into format understandable by BHS